jerry-core/parser/js/js-lexer.c

```diff
@@ -917,6 +917,28 @@ lexer_parse_string (parser_context_t *context_p) /**< context */
   context_p->column = (parser_line_counter_t) (column + 1);
 } /* lexer_parse_string */
 
+/**
+ * Parse octal number.
+ */
+static inline void
+lexer_parse_octal_number (parser_context_t *context_p, /** context */
+                          const uint8_t **source_p) /**< current source position */
+{
+  do
+  {
+    (*source_p)++;
+  }
+  while (*source_p < context_p->source_end_p
+         && *source_p[0] >= LIT_CHAR_0
+         && *source_p[0] <= LIT_CHAR_7);
+
+  if (*source_p < context_p->source_end_p
+      && (*source_p[0] == LIT_CHAR_8 || *source_p[0] == LIT_CHAR_9))
+  {
+    parser_raise_error (context_p, PARSER_ERR_INVALID_NUMBER);
```

Comment:
Shouldn't we use the `PARSER_ERR_INVALID_OCTAL_DIGIT` error here as well?

Reply:
Yes, that is more precise error message. I've modified the PR.